python/ray/tests/test_scheduling.py

```diff
@@ -674,6 +675,48 @@ def train(self):
     ray.get(trainer.train.remote(), timeout=30)
 
 
+@pytest.mark.parametrize(
+    "ray_start_regular", [{
+        "_system_config": {
+            "metrics_report_interval_ms": 1000,
+            "complex_scheduling_class": True
+        }
+    }],
+    indirect=True)
+def test_scheduling_class_depth(ray_start_regular):
+
+    node_info = ray.nodes()[0]
+    metrics_export_port = node_info["MetricsExportPort"]
+    addr = node_info["NodeManagerAddress"]
+    prom_addr = f"{addr}:{metrics_export_port}"
+
+    @ray.remote(num_cpus=1000)
+    def infeasible():
+        pass
+
+    @ray.remote
+    def start_infeasible():
+        ray.get(infeasible.remote())
+
+    start_infeasible.remote()
+    infeasible.remote()
+
+    # We expect the 2 calls to `infeasible` to be separate scheduling classes
+    # because one has depth=1, and the other has depth=2.
+
+    metric_name = "ray_internal_num_infeasible_scheduling_classes"
+
+    def condition():
+        _, metric_names, metric_samples = fetch_prometheus([prom_addr])
```

Comment:
nice test! This is a great pattern we can use in the future for other challenging cases.